Neo4j reader

coyotebush opened the pull request:
Work in progress. Since I'm completely new to Spring, I don't know whether I'm doing anything correctly, but I'm roughly following the [Hubway code](https://github.com/jexp/hubway-sdn).

I'm not even sure how to try it out -- what needs to be done to set up Spring and then call `importStations`, for instance?

If the parsing tweaks look good, I could create another pull request to go ahead and merge just those.

coyotebush committed:
Convert a parser.data.TripData to a domain.Trip

coyotebush committed:
Connect reader to parser and Neo4j

coyotebush committed:
Add Bike and Station nodes and relationships as necessary

coyotebush commented:
While we're at it... probably the Bike/Station/Trip classes should be under a "graph" package or something, to avoid possible conflict with similar classes on the RDBMS side?

mhotan commented on graph-db-driver/src/main/java/se/kth/csc/moderndb/cbexplorer/parser/TripReader.java:
```diff
@@ -1,7 +0,0 @@
-/**
- * Created by mhotan on 4/13/14.
- */
-public class TripReader {
```
I was thinking the TripReader class was going to be a Spring Controller like class.  Then we could Autowire A GraphService class.  But I think we have to look into how we will define the applicationContext.xml (Spring thing) for this particular Restful service.  The Application Context will define how things are auto wired.

mhotan commented:
Nothing stands out to me.  Its going to be a lot of experimentation to figure out how to make this work with the framework.

coyotebush committed:
Refactor TripReader and CitiBikeGraphService

coyotebush committed:
Merge branch 'neo4j-repo' into graph-reader

mhotan committed:
Initial working parse and loader for an Embedded database.

Need to change the Configuration to use a standalone neo4j server.

mhotan committed:
Initial implementation of Embedded CitiBike Graph

Currently there are two modules.  graph-load-db is used to load the data into a local embedded database.  Currently there are stability issues with Spring causing a lot of duplication with the second module graph-db-driver.  Which is currently having problems a REST service that interacts with the embedded DB

mhotan committed:
Merge branch 'graph-reader' of https://github.com/mhotan/DD2471 into graph-reader-base

Conflicts:
	graph-db-driver/build.gradle
	graph-db-driver/src/main/java/se/kth/csc/moderndb/cbexplorer/domain/CitiBikeGraphService.java

mhotan committed:
Updated README

coyotebush committed:
Remove CitiBikeGraphService and TripReader

This functionality is now in Application in graph-load-db

coyotebush commented:
Looking good, thanks! The import does run quite slowly -- I'm seeing somewhere ~10 trips/second. I wonder if explicitly controlling transactions might affect performance?

I think we should probably merge this soon and leave the REST API for next?

mhotan commented:
Yeah we can merge this and do REST API and make the import faster next.
